spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/AdminController.java

```diff
@@ -122,6 +125,38 @@
         chartColorsPerHttpResponseCode = Collections.unmodifiableMap(colorMap);
     }
 
+    private static final int oneMinuteToMilliseconds = 60000;
+    private static final int criticalThresholdInMinutes = 60;
+    private static final int warningThresholdInMinutes = 5;
+
+    private String gatewayHealthStatusSummary(){
+        List<Proxy> proxiesList = proxyRepository.findAvailable();
+        if (proxiesList.stream().map(Proxy::getStatus).anyMatch(status -> status == ProxyStates.CRITICAL))
+            return "Critical";
+        else if(proxiesList.stream().map(Proxy::getStatus).anyMatch(status -> status == ProxyStates.WARNING))
+            return "Warning";
+        return "Ok";
+    }
+
+    private void updateProxyInfoForGatewayHealthMonitor() {
```

Comment:
I would suggest to just save the current time here.

Reply:
Why save the current time?
The current time is only used to compute how many minutes ago was the last sucessful call made to admin

Reply:
This will be changed so I consider this solved